includes/sanitizers/class-amp-form-sanitizer.php

```diff
@@ -58,32 +58,8 @@ public function sanitize() {
 				$node->setAttribute( 'method', $method );
 			}
 
-			/*
-			 * In HTML, the default action is just the current URL that the page is served from.
-			 * The action "specifies a server endpoint to handle the form input. The value must be an
-			 * https URL and must not be a link to a CDN".
-			 */
-			if ( ! $node->getAttribute( 'action' ) ) {
-				$action_url = esc_url_raw( '//' . $_SERVER['HTTP_HOST'] . wp_unslash( $_SERVER['REQUEST_URI'] ) );
-			} else {
-				$action_url = $node->getAttribute( 'action' );
-
-				// Handle relative URLs.
-				if ( ! preg_match( '#^(https?:)?//#', $action_url ) ) {
-					$schemeless_host = '//' . $_SERVER['HTTP_HOST'];
-					if ( '?' === $action_url[0] || '#' === $action_url[0] ) {
-						// For actions consisting of only a query or URL fragment, include the schemeless-host and the REQUEST URI of the current page.
-						$action_url = $schemeless_host . wp_unslash( $_SERVER['REQUEST_URI'] ) . $action_url;
-					} elseif ( '.' === $action_url[0] ) {
```

Comment:
In your changes, you correctly changed the `'.' === $action_url[0]` to `'/' !== $action_url[0]`. This I believe fixes the case where there is an `action` URL like `foo/` as opposed to `../`. I added a test to confirm this: 585be89e054a1009a561e36904bfa51d1fe18361.